### PR TITLE
xcm-executor: Improve logging

### PR DESCRIPTION
**Pull request by bkchr**
This improves logging in the xcm-executor to have better debugability when executing a XCM message.

### CODE REVIEW

File: polkadot/xcm/xcm-executor/src/traits/should_execute.rs [hunk 1]
```diff
@@ -59,19 +59,36 @@ impl ShouldExecute for Tuple {
...
+					tracing::trace!(
+						target: "xcm::should_execute",
+						?origin,
+						?instructions,
+						?max_weight,
+						?properties,
+						?error,
+						%barrier,
+						"did not pass barrier",
+					);
```
Comment (acatangiu) [inline polkadot/xcm/xcm-executor/src/traits/should_execute.rs:87]: I tried using this, but for some reason this never gets printed 🤷‍♂️ too late in the day now to investigate why

Comment (bkchr) [reply polkadot/xcm/xcm-executor/src/traits/should_execute.rs:87]: Hmm.

Should work with https://github.com/paritytech/polkadot-sdk/pull/4863

**Comment by bkchr** [comment]
bot fmt

**Comment by bkchr** [comment]
bot fmt

File: polkadot/xcm/xcm-executor/src/traits/should_execute.rs (hunk 1 above)
Comment (acatangiu) [reply polkadot/xcm/xcm-executor/src/traits/should_execute.rs:86]: ah, yes, I was missing #4863 on my branch, works now, thanks!

**Review by acatangiu — APPROVED** [review]

**Review by franciscoaguirre — APPROVED** [review]

**Pull request merged**